compiler/noirc_evaluator/src/ssa/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -1643,6 +1662,28 @@ impl<F: AcirField> AcirContext<F> {
         Ok(())
     }
 
+    /// Recursively create acir values for returned arrays. This is necessary because a brillig returned array can have nested arrays as elements.
+    /// A singular array of witnesses is collected for a top level array, by deflattening the assigned witnesses at each level.
```

Comment:
```suggestion
    /// Recursively create zeroed-out acir values for returned arrays. This is necessary because a brillig returned array can have nested arrays as elements.
```